core/src/main/scala/scala/macros/core/Universe.scala

```diff
@@ -92,5 +98,18 @@ trait Universe {
   // Expansion
   // =========
   type Expansion
+  def enclosingPosition: Position
+  def enclosingOwner: Symbol
+}
+
+trait Input extends Any {
+  def path: Path
+}
 
+trait Position extends Any {
+  def start: Int
+  def end: Int
+  def line: Int
+  def column: Int
+  def input: Input
 }
```

Comment:
`Position` seems can be defined outside of `Universe`, if we simplify `input: Input` to `path: String`?

Reply:
Position is already defined outside of universe, regardless of Input.

Reply:
I just see there's a Position in `core/src/main/scala/scala/macros/inputs`.  It seems that directory can be removed. But that's for cleanup in another PR.